client.go

```diff
@@ -450,3 +450,16 @@ func (c *client) sendQuery(msg *dns.Msg) error {
 	}
 	return nil
 }
+
+// Browsev3 - A new type of browser which keeps a local cache of network services
+//    As events happen in network, including TTL expiry or new host joins, the event is returned in the channel
+func (r *Resolver) Browsev3(ctx context.Context, service, domain string, events chan<- *ServiceEvent) (*Browser, error) {
```

Comment:
To be honest, I'm not happy with adding a separate API function.
Though, we can use this approach as an intermediate solution. 
Finally, there should be only one single Browse method.